tests/integration/targets/mso_schema_template_bd/tasks/main.yml

```diff
@@ -64,35 +64,43 @@
     - '{{ mso_site | default("ansible_test") }}'
     state: present
 
-- name: Ensure schema 1 with Template 1 exist
+- name: Ensure schema 1 with  exist
   mso_schema_template: &schema_present
     <<: *mso_info
     schema: '{{ mso_schema | default("ansible_test") }}'
     tenant: ansible_test
-    template: Template 1
+    template: Template1
     state: present
 
-- name: Ensure schema 1 with Template 2 exist
+- name: Ensure schema 1 with Template2 exist
   mso_schema_template:
     <<: *schema_present
     schema: '{{ mso_schema | default("ansible_test") }}'
     tenant: ansible_test
-    template: Template 2
+    template: Template2
     state: present
 
-- name: Ensure schema 2 with Template 3 exist
+- name: Ensure schema 2 withTemplate3 exist
   mso_schema_template:
     <<: *schema_present
     schema: '{{ mso_schema | default("ansible_test") }}_2'
     tenant: ansible_test
-    template: Template 3
+    template: Template3
+    state: present
```

Comment:
Since you are referring to *schema_present, state: present is already there. you might be rewriting that. Same for next task as well